code/obj/critter/drone.dm

```diff
@@ -1325,3 +1325,186 @@
 				src.health = 0
 				src.CritterDeath()
 
+/obj/critter/gunbot/drone/buzzdrone/sawfly // For associated objects check sawfly.dm in the obj folder
+
+
+	name = "Sawfly"
+	desc = "A folding antipersonnel drone of syndicate origin. It'd be pretty cute if it wasn't trying to kill people."
+	icon_state = "sawfly"
+	beeptext = "UH OOH"
+	dead_state = "sawflydead"
+	projectile_type = /datum/projectile/laser/drill/sawfly
+	current_projectile = new/datum/projectile/laser/drill/sawfly
+	//angertext = "detects the presence of"
+	smashes_shit = 0
+	droploot = null //change this later
+	health = 40
+	maxhealth = 40
+	firevuln = 0.5
+	brutevuln = 1.2
+	can_revive = 1
+	atksilicon = 0
+	firevuln = 0
+	atk_delay = 5
+	attack_cooldown = 8
+	seekrange = 15
+	var/deathtimer = 0 // for catastrophic failure on death
+	var/isnew = TRUE // for reuse
+	var/sawflynames = list("A", "B", "C", "D", "E", "F", "V", "W", "X", "Y", "Z", "Alpha", "Beta", "Gamma", "Lambda", "Delta")
+
+	proc/foldself()
+		var/obj/item/old_grenade/spawner/sawfly/reused/N = new /obj/item/old_grenade/spawner/sawfly/reused(get_turf(src))
+		// pass our name and health
+		N.name = "Compact [name]"
+		N.tempname = src.name
+		N.temphp = src.health
+		qdel(src)
+
+	proc/blowup() // used in emagged controllers and has a chance to activate when they die
+
+		if(prob(66))
+			src.visible_message("<span class='combat'>[src]'s [pick("motor", "core", "fuel tank", "battery", "thruster")] [pick("combusts", "catches on fire", "ignites", "lights up", "bursts into flames")]!")
+			fireflash(src,1,TRUE)
+
+		else
+			src.visible_message("<span class='combat'>[src]'s [pick("motor", "core", "head", "engine", "thruster")] [pick("overloads", "blows up", "catastrophically fails", "explodes")]!")
+			fireflash(src,0,TRUE)
+			explosion(src, get_turf(src), 0, 1, 2, 3)
+			qdel(src)
+
+		if(alive) // prevents weirdness from emagged controllers
+			qdel(src)
+
+	emp_act() //same thing as if you emagged the controller, but much higher chance
+		if(prob(80))
+			src.visible_message("<span class='combat'>[src] buzzes oddly and starts to sprial out of contro!</span>")
+			SPAWN(2 SECONDS)
+				src.blowup()
+		else
+			src.foldself()
+
+
+
+	New()
+		..()
+		deathtimer = rand(1, 5)
+		death_text = "[src] jutters and falls from the air, whirring to a stop"
```

Comment:
any reason this is in New?

Reply:
due to them being non-constant functions, they have to be declared in new, since it doesn't like them being in the normal setup